pkg/controller/common/rollout/workloads/controller.go

```diff
@@ -104,3 +105,39 @@ func (c *cloneSetController) fetchCloneSet(ctx context.Context) error {
 	c.cloneSet = &workload
 	return nil
 }
+
+// deploymentController is the place to hold fields needed for handle Deployment type of workloads
+type deploymentController struct {
+	workloadController
+	targetNamespacedName types.NamespacedName
+	deploy               *appsv1.Deployment
+}
+
+// size fetches the Deloyment and returns the replicas (not the actual number of pods)
+func (c *deploymentController) size(ctx context.Context) (int32, error) {
+	if c.deploy == nil {
+		err := c.fetchDeployment(ctx)
+		if err != nil {
+			return 0, err
+		}
+	}
+	// default is 1
+	if c.deploy.Spec.Replicas == nil {
+		return 1, nil
+	}
+	return *c.deploy.Spec.Replicas, nil
+}
+
+func (c *deploymentController) fetchDeployment(ctx context.Context) error {
+	// get the deployment
+	workload := appsv1.Deployment{}
+	err := c.client.Get(ctx, c.targetNamespacedName, &workload)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			c.recorder.Event(c.parentController, event.Warning("Failed to get the Deployment", err))
+		}
+		return err
+	}
+	c.deploy = &workload
+	return nil
+}
```

Comment:
Why are those here? Looks like they are only used by the deploymentScaler

Reply:
Fixed. I merged these helper functions into DeploymentScaleController.